File: Java/128_Longest_Consecutive_Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> values = new HashSet<>();

        for(int num : nums) {
            //put all values in a set
            values.add(num);
        }

        int max = 0;
        for(int num : values) {
            //only check if we are at the start of a sequence (nothing before it)
            if(!values.contains(num - 1)) {
                int length = 0;
                //then go through and check for how long an ascending value exists
                while(values.contains(num + length)) {
                    length++;
                }
                //see if we got a new maximum
                max = Math.max(length, max);
            }
        }
        return max;
    }
}

File: Python/128_Longest_Consecutive_Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        vals = set(nums)
        seq = 0
        for val in vals:
            if val - 1 not in vals:
                count = 1
                while val + count in vals:
                    count += 1
                seq = max(seq, count)
                
        return seq
